cdap-support-bundle/bin/observability.py

```diff
@@ -0,0 +1,539 @@
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_logs
+        },
+        {
+            "command": f'kubectl logs -l cdap.twill.app=system-worker -n {namespace}',
+            "fileName": 'kubectl-log-system-worker',
+            "fileExt": 'log',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_logs
+        },
+        {
+            "command": f'kubectl logs -l cdap.twill.app=task-worker -n {namespace} -c taskworkertwillrunnable',
+            "fileName": 'kubectl-log-task-worker-c-taskworkertwillrunnable',
+            "fileExt": 'log',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_logs
+        },
+        {
+            "command": f'kubectl logs -l cdap.twill.app=task-worker -n {namespace} -c artifactlocalizertwillrunnable',
+            "fileName": 'kubectl-log-task-worker-c-artifactlocalizertwillrunnable',
+            "fileExt": 'log',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_logs
+        },
+        {
+            "command": f'kubectl logs -l cdap.twill.app=preview-runner -n {namespace} -c artifactlocalizertwillrunnable',
+            "fileName": 'kubectl-log-preview-runner-c-artifactlocalizertwillrunnable',
+            "fileExt": 'log',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_logs
+        },
+        {
+            "command": f'kubectl logs -l cdap.twill.app=preview-runner -n {namespace} -c previewrunnertwillrunnable',
+            "fileName": 'kubectl-log-preview-runner-c-previewrunnertwillrunnable',
+            "fileExt": 'log',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": cdap_services_logs
+        },
+        {
+            "command": f'kubectl exec -ti {instance_name}-fs-0 -n {namespace} -- /usr/local/hadoop/bin/hdfs dfs -df -h',
+            "fileName": 'kubectl-exec-ti-cdap-fs-0-n-usr-local-hadoop-dfs-HDFS-disk-usage',
+            "fileExt": 'log',
+            "fileWrite": True,
+            "forEachNode": False,
+            "forEachCommand": '',
+            "placeholderString": "",
+            "parentFolder": memory_logs
+        },
+        {
+            "command": f'kubectl exec cdap-{instance_name}-runtime-0 -n {namespace} -- curl cdap-{instance_name}-router:11015/v3/system/services/appfabric/logs?start=0 -H "{auth_token}" ',
```

Comment:
what happens when auth is disabled? Does this still work as expected?

Reply:
> To get instance name - kubectl get cdapmasters.cdap.cdap.io -o yaml -o=jsonpath='{.items[0].metadata.namespace}'
> 
> k8s namespace: kubectl get cdapmasters.cdap.cdap.io -o yaml -o=jsonpath='{.items[0].metadata.name}'

get instance is working fine but namespace command will only work if your selected namespace is hdf runtime Namespace. In case of default or any other namespace it will fail.